compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -2233,3 +2646,24 @@ fn find_param_in_ty<'tcx>(ty: Ty<'tcx>, param_to_point_at: ty::GenericArg<'tcx>)
     }
     false
 }
+
+fn find_param_def_in_ty_walker<'tcx>(
```

Comment:
There's a small convenience thing which is in most cases I have a `GenericParamDef` but `find_param_in_ty` wants a `GenericArg`; so I was just avoiding typing out `ty::ParamTy::for_def(...)` a few times.

The other (bigger) issue was that it expects a `ty::Ty`, but I frequently instead have a `ty::GenericArg`, which is sorta like a supertype of `ty::Ty` since it's either a `Ty` or a constant or a lifetime. Ideally I think it would ask for like a `impl TypeWalkable` since both `ty::Ty` and `ty::GenericArg` have `walk(self) -> TypeWalker`, but that's just a pun and there's no actual shared trait between the two.

Asking for the `TypeWalker` instead of the thing being walked over effectively gives "polymorphism" for free over the two types, then.

Potentially I could just update the existing `find_param_in_ty` to instead ask for a `TypeWalker`? Or I could write a local trait just for this helper function to allow `walk()` to be called generically. Or I could just do the unwrapping at each place to convert `GenericArg` into `Ty`; I'm not exactly how much boilerplate (if any) that would add.

Reply:
Alternatively, I'm pretty sure that `GenericArg` implements `From<Ty>`, and you could just make `find_param_in_ty` take a `GenericArg`, and update the other call sites to just do an `ty.into()`.